python312Packages.aiokafka: 0.10.0 -> 0.11.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for python312Packages.aiokafka is: Kafka integration with asyncio

meta.homepage for python312Packages.aiokafka is: https://aiokafka.readthedocs.org

meta.changelog for python312Packages.aiokafka is: https://github.com/aio-libs/aiokafka/releases/tag/v0.11.0


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 0.11.0 with grep in /nix/store/4cc4mh03j9hh237pzh455fsiwz5vclv8-python3.12-aiokafka-0.11.0
- found 0.11.0 in filename of file in /nix/store/4cc4mh03j9hh237pzh455fsiwz5vclv8-python3.12-aiokafka-0.11.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
4 total rebuild path(s)

4 package rebuild(s)

First fifty rebuilds by attrpath

home-assistant-component-tests.apache_kafka
python311Packages.aiokafka
python312Packages.aiokafka
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/4cc4mh03j9hh237pzh455fsiwz5vclv8-python3.12-aiokafka-0.11.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python312Packages.aiokafka https://github.com/r-ryantm/nixpkgs/archive/48a606941f76a500bdfe57fcce333936102fe702.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/48a606941f76a500bdfe57fcce333936102fe702#python312Packages.aiokafka
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/4cc4mh03j9hh237pzh455fsiwz5vclv8-python3.12-aiokafka-0.11.0
ls -la /nix/store/4cc4mh03j9hh237pzh455fsiwz5vclv8-python3.12-aiokafka-0.11.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.apache_kafka</li>
    <li>python311Packages.aiokafka</li>
    <li>python311Packages.aiokafka.dist</li>
    <li>python312Packages.aiokafka</li>
    <li>python312Packages.aiokafka.dist</li>
  </ul>
</details>

---

###### Maintainer pings

dotlambda commented on pkgs/development/python-modules/aiokafka/default.nix:
```diff
@@ -17,7 +17,7 @@
+  version = "0.11.0";
   pyproject = true;
 
   disabled = pythonOlder "3.7";
```
```suggestion
  disabled = pythonOlder "3.8";
```

dotlambda reviewed (commented):
Should depend on typing-extensions rather than python-kafka.

dotlambda committed:
python312Packages.aiokafka: refactor

Diff: https://github.com/aio-libs/aiokafka/compare/refs/tags/v0.10.0...v0.11.0

Changelog: https://github.com/aio-libs/aiokafka/releases/tag/v0.11.0

dotlambda committed:
python312Packages.kafka-python: drop

The only users aiokafka and parsedmarc stopped depending on it.

natsukium commented:
Result of `nixpkgs-review pr 328348` at 3153d4f5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>home-assistant-component-tests.apache_kafka</li>
    <li>python311Packages.aiokafka</li>
    <li>python312Packages.aiokafka</li>
  </ul>
</details>
